crates/ethereum/evm/src/lib.rs

```diff
@@ -45,33 +45,71 @@ pub mod dao_fork;
 pub mod eip6110;
 
 /// Ethereum EVM implementation.
-#[derive(derive_more::Debug, derive_more::Deref, derive_more::DerefMut, derive_more::From)]
+#[derive(derive_more::Debug)]
 #[debug(bound(DB::Error: Debug))]
-pub struct EthEvm<'a, EXT, DB: Database>(reth_revm::Evm<'a, EXT, DB>);
+pub struct EthEvm<DB: Database> {
+    db: Option<DB>,
+    evm_env: EvmEnv,
+}
+
+impl<DB: Database> EthEvm<DB> {
+    /// Creates a new EVM instance.
+    pub fn new(db: DB, evm_env: EvmEnv) -> Self {
+        Self { db: Some(db), evm_env }
+    }
+}
 
-impl<EXT, DB: Database> Evm for EthEvm<'_, EXT, DB> {
+impl<DB: Database> Evm for EthEvm<DB> {
     type DB = DB;
     type Tx = TxEnv;
     type Error = EVMError<DB::Error>;
 
     fn block(&self) -> &BlockEnv {
-        self.0.block()
+        &self.evm_env.block_env
     }
 
     fn into_env(self) -> EvmEnv {
-        let Env { cfg, block, tx: _ } = *self.0.context.evm.inner.env;
-        EvmEnv {
-            cfg_env_with_handler_cfg: CfgEnvWithHandlerCfg {
-                cfg_env: cfg,
-                handler_cfg: self.0.handler.cfg,
-            },
-            block_env: block,
-        }
+        self.evm_env
     }
 
     fn transact(&mut self, tx: Self::Tx) -> Result<ResultAndState, Self::Error> {
-        *self.tx_mut() = tx;
-        self.0.transact()
+        let db = self.db.take().expect("db is set");
+
+        let mut evm = EvmBuilder::default()
+            .with_db(db)
+            .with_cfg_env_with_handler_cfg(self.evm_env.cfg_env_with_handler_cfg.clone())
+            .with_block_env(self.evm_env.block_env.clone())
+            .with_tx_env(tx)
+            .build();
+
+        let res = evm.transact();
```

Comment:
I think the Rise guys had said that re-instantiating an EVM each time can be expensive execution-wise, no?